scipy/optimize/_linprog_rs.py

```diff
@@ -341,7 +340,7 @@ def _phase_two(c, A, x, b, maxiter, tol, maxupdate, mast, pivot, iteration=0,
             else:
                 phase = 2
                 x_postsolve = x
-            x_o, fun, slack, con, _, _ = _postsolve(x_postsolve, *_T_o,
+            x_o, fun, slack, con, _, _ = _postsolve(x_postsolve.copy(), *_T_o,
```

Comment:
It feels like copying `x_postsolve` is  *every time* is passed to `_postsolve` is going to make this a fair bit slower. I've had a look through `_postsolve` and we're already copying `x` if there are variable substitutions (``np.array(x)``) makes a copy by default.

https://github.com/scipy/scipy/blob/368e211cf3a37351f8651cea4dd8ed3c884f9f44/scipy/optimize/_linprog_util.py#L1185-L1190

Maybe unindenting line 1190 would be better? That way ``x`` is copied once instead of twice if variables are substitued.

Reply:
I decided not to look into _postprocess at all once I found the issue. I wanted to avoid the potential to add a new bug in haste to get this backported. I figured callbacks and display don't lend themselves to speedy execution anyone. I thought about adding this to the list of maintenance. Let's do that instead?

Reply:
Do we have any estimates of how often ``len(undo) > 0``? If it's true for most/enough cases `x` will be copied on line 1190 anyway and unindenting should result in a copy for all cases.

I agree that they don't lend themselves to speedy execution, but, I would still prefer to avoid unnecessary work if possible.

Reply:
`undo` is used when variables can be eliminated in presolve. (The removal is 'undone' in the postprocessing snippet above.) This is possible when there are trivial conditions like row singletons or zero columns in the constraint matrix. I don't know how often it happens in practice, though.
Anyway, hope take a closer look soon.

Reply:
Didn't want to slow down interior-point unnecessarily when `undo` isn't used, so I added a copy option to `_postsolve` and use it in revised simplex.

Reply:
LGTM :+1: I'll mark this as resolved.